### ISSUE TITLE
DataSource.getByRange (toArray = true) doesn't work for nested objects.

### ISSUE THREAD

**Issue by lewisjb** [handsontable/handsontable#3959]
### Description
Line 159 in src/dataSource.js
https://github.com/handsontable/handsontable/blob/master/src/dataSource.js#L159
(`newRow.push(row[prop]);`)

Won't/doesn't work for https://docs.handsontable.com/0.29.1/tutorial-data-sources.html#example5

```javascript
nestedObjects = [
  {id: 1, name: {first: "Ted", last: "Right"}, address: ""},
  {id: 2, address: ""}, // HOT will create missing properties on demand
  {id: 3, name: {first: "Joan", last: "Well"}, address: ""}
]
```
`row[prop]` will result in `row['name.first']`, which doesn't exist.

The line should be using `getProperty` (https://github.com/handsontable/handsontable/blob/master/src/helpers/object.js#L272).

**Comment by AMBudnik** [comment]
Hi,

Thanks for sharing this Feature Request. I am moving it to our Wiki Section at https://github.com/handsontable/handsontable/wiki/Feature-requests
The issue will stay closed until it'll get picked by our developer.

**Comment by lewisjb** [comment]
@AMBudnik This isn't a feature request, it's a bug (that is linked to the #3958 bug)

**Comment by AMBudnik** [comment]
Sorry @lewisjb 
I misunderstood the case. I'm changinf the label.
Thanks for notifying

**Comment by wojciechczerniak** [comment]
Duplicate of #6548

**Comment by AMBudnik** [comment]
Referenced issue https://github.com/handsontable/handsontable/issues/3958 is fixed in v 8.0.0;
